src/state_machine.zig

```diff
@@ -31,11 +31,83 @@ pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
         const GrooveType = @import("lsm/groove.zig").GrooveType;
         const ForestType = @import("lsm/forest.zig").ForestType;
 
-        const AccountsGroove = GrooveType(
+        pub const AccountImmutable = extern struct {
+            id: u128,
+            user_data: u128,
+            timestamp: u64,
+            ledger: u32,
+            code: u16,
+            flags: AccountFlags,
+            padding: [16]u8,
```

Comment:
Originally, Immutable had a larger `reserved` to account for the padding but keeping them smaller made more sense to reduce space overhead.

Reply:
But if in the future we want to use all of `reserved`, won't we need to put it somewhere?

Reply:
We would then put it into Immutable/Mutable as applicable, updating the padding to account (no pun intended) for the new fields. I assume `reserved` serves as both padding and type-erased implementation storage for Transfer/Account, where Immutable/Mutable would *be* the implementation with fields more specific than `reserved`.

Reply:
Ok, but unless my math is wrong we are 8 bytes short. Adding those changes the disk format, so it is something we want to accommodate in advance.

Reply:
Had some more time to think about this. Adding 8 more bytes to `AccountImmutable.padding` would no longer make the struct a power of two size (which is comptime asserted down the line by `SetAssociativeCache`). Also, I think the other 8 bytes are accounted for with `AccountMutable` having a copy of `timestamp` (vs. a single Account struct having only one `timestamp`)

Reply:
> Had some more time to think about this. Adding 8 more bytes to AccountImmutable.padding would no longer make the struct a power of two size (which is comptime asserted down the line by SetAssociativeCache).

:+1: Ok, that's fine then.  But just fyi:

> Also, I think the other 8 bytes are accounted for with AccountMutable having a copy of timestamp (vs. a single Account struct having only one timestamp)

Since those 8 bytes are "duplicated", they don't actually solve the problem. We still wind up with 8 bytes of `reserved` that we don't have room to store. But its fine for now.